tests/integration/test_charm_with_profile.py

```
@@ -0,0 +1,135 @@
+
+
+def apply_manifests(lightkube_client: lightkube.Client, yaml_file_path: str):
+    """Apply resources using manifest files and returns the applied object.
+
+    Args:
+        lightkube_client (lightkube.Client): an instance of lightkube.Client to
+            use for applying resources.
+        yaml_file_path (str): the resource yaml file.
+
+    Returns:
+        A namespaced or global lightkube resource (obj).
+    """
+    read_yaml = Path(yaml_file_path).read_text()
+    yaml_loaded = codecs.load_all_yaml(read_yaml)
+    for obj in yaml_loaded:
+        try:
+            lightkube_client.apply(
+                obj=obj,
+                name=obj.metadata.name,
+            )
+        except lightkube.core.exceptions.ApiError as api_error:
+            raise api_error
+    return obj
+
+
+@pytest.fixture(scope="module")
+def lightkube_client() -> lightkube.Client:
+    """Returns a lightkube Client that can talk to the K8s API."""
+    client = lightkube.Client(field_manager="kfp-operators")
+    return client
+
+
+@pytest.fixture(scope="module")
+def apply_profile(lightkube_client):
+    """Apply a Profile simulating a user."""
+    # Create a Viewer namespaced resource
+    profile_class_resource = create_global_resource(  # noqa F841
+        group="kubeflow.org", version="v1", kind="Profile", plural="profiles"
+    )
+
+    # Apply Profile first
+    apply_manifests(lightkube_client, PROFILE_FILE_PATH)
+
+    # Allow time for the Profile to be created
+    time.sleep(10)
+
+    yield
+
+    # Remove namespace
+    read_yaml = Path(PROFILE_FILE_PATH).read_text()
+    yaml_loaded = codecs.load_all_yaml(read_yaml)
+    for obj in yaml_loaded:
+        try:
+            lightkube_client.delete(
+                res=type(obj),
+                name=obj.metadata.name,
+                namespace=obj.metadata.namespace,
+            )
+        except lightkube.core.exceptions.ApiError as api_error:
+            raise api_error
```

Same observation as above, can't we just let the error be raised here since we're not handling it in any special way?
```suggestion
        lightkube_client.delete(
            res=type(obj),
            name=obj.metadata.name,
            namespace=obj.metadata.namespace,
        )
```